src/Dhall/Parser.hs

```diff
@@ -101,26 +91,34 @@ instance Monoid a => Monoid (Parser a) where
     mappend = liftA2 mappend
 
 instance IsString a => IsString (Parser a) where
-    fromString x = fmap fromString (Text.Parser.Char.string x)
+    fromString x = fromString x <$ Text.Megaparsec.Char.string (fromString x)
+
+instance Text.Parser.Combinators.Parsing Parser where
+  unexpected = fail
+  eof = Parser Text.Megaparsec.eof
+  try = Text.Megaparsec.try
+  (<?>) = (Text.Megaparsec.<?>)
+  notFollowedBy = Text.Megaparsec.notFollowedBy
+
+instance Text.Parser.Char.CharParsing Parser where
+  satisfy = Parser . Text.Megaparsec.Char.satisfy
```

Comment:
You will probably get a speed improvement if you implement the `text` method of the `CharParsing` class with a high-performance primitive (and probably the rest of the class, too).  One of the reasons for using `megaparsec` is that it provides bulk parsing primitives

The other reason I suggest this is because that is where the parsing bottleneck is on `master`